### PR TITLE
CA-387698: datamodel: eliminate next_release

### PR DESCRIPTION
**Pull request by robhoes**
Replace all occurances of `next_release` and `rel_next` with appropriate version numbers.

### CODE REVIEW

File: ocaml/idl/schematest.ml [hunk 1]
```diff
@@ -3,7 +3,7 @@ let hash x = Digest.string x |> Digest.to_hex
 (* BEWARE: if this changes, check that schema has been bumped accordingly in
    ocaml/idl/datamodel_common.ml, usually schema_minor_vsn *)
```
Comment (psafont) [inline ocaml/idl/schematest.ml:4]: image

Comment (robhoes) [reply ocaml/idl/schematest.ml:4]: I did..............(though not strictly needed here)? 🤨

**Review by edwintorok — COMMENTED** [review]
This is good, but how are we going to introduce new parameters now? We'll need to specify a version, but you don't have a version until you tag it.
Do we need something equivalent to the 'prototyped' lifecycle that we have for APIs, but for API named parameters?

File: ocaml/idl/datamodel_common.ml [hunk 2]
```diff
@@ -349,9 +349,9 @@ let get_product_releases in_product_since =
   in
   go_through_release_order release_order
 
-let next_release =
+let numbered_release version =
```
Comment (lindig) [inline ocaml/idl/datamodel_common.ml:352]: I would have called this just "release"; there is no check that this is indeed numbered or using any particular format. Which could be done with a regular expression or Scanf.

**Review by lindig — APPROVED** [review]

**Comment by robhoes** [comment]
>This is good, but how are we going to introduce new parameters now? We'll need to specify a version, but you don't have a version until you tag it.

We still need to automated this for parameters, like we do for fields and functions. Until then, I think the best option is to manually put something like `1.2.3-next` like the automation would do. At least then we have more info to fix things up then in the old case, which would make it just `next`.

File: ocaml/idl/datamodel_common.ml (hunk 2 above)
Comment (robhoes) [reply ocaml/idl/datamodel_common.ml:352]: It's just to distinguish this case from the old releases, which have names.

**Comment by psafont** [comment]
> I think the best option is to manually put something like 1.2.3-next

What's the best place to document this?

**Comment by robhoes** [comment]
Perhaps I'll make that `numbered_release` accept only `x.y.z` and `x.y.z-next`, and raise an error from there explaining the rules.

**Comment by psafont** [comment]
Conflicts are because the lifecycle got updated incorrectly, we should prioritise to merge this PR

**Review by psafont — APPROVED** [review]

**Pull request merged**